app/views/directives/osc-persistent-volume-claim.html

```diff
@@ -2,37 +2,43 @@
   <fieldset ng-disabled="claimDisabled">
     <div ng-show="storageClasses | hashSize" class="form-group">
       <!--storage class-->
-      <label class="required">Storage Classes</label>
-      <div id="claim-storage-class-help" class="help-block mar-bottom-lg">
+      <label>Storage Class</label>
+      <div>
+      <ui-select
+        ng-if="storageClasses"
+        ng-model="claim.storageClass"
+        theme="bootstrap"
+        search-enabled="true"
+        title="Select a storage class"
+        class="select-role"
+        flex>
+        <ui-select-match placeholder="Select a storage class">
+          <span>
+            {{$select.selected.metadata.name}}
+          </span>
+        </ui-select-match>
+        <ui-select-choices
+          repeat="sclass as sclass in storageClasses | toArray | filter : { metadata: { name: $select.search } } | orderBy: 'metadata.name'">
+          <div>{{ sclass.metadata.name }}
+            <span ng-if="sclass | annotation : 'description'" class="text-muted">
+              <small>&ndash;
+                <span ng-if="sclass.parameters.type">Type: {{sclass.parameters.type}} | </span>
+                <span ng-if="sclass.parameters.zone">Zone: {{sclass.parameters.zone}} | </span>
+                {{sclass | annotation : 'description'}}
+              </small>
+            </span>
+          </div>
+        </ui-select-choices>
+      </ui-select>
+      </div>
+      <div id="claim-storage-class-help" class="help-block">
         Storage classes are set by the administrator to define types of storage the users can select.
+        <span ng-if="defaultStorageClass">  If not set, default storage class <var>{{defaultStorageClass.metadata.name}}</var> will be used.</span>
```

Comment:
If I'm reading the code right, there's no way to NOT set a storage class when there's a default since there's no empty option. In which case we don't need this message. But we should make sure the default is initially selected in the controller.